### PR TITLE
[CLOUD-2262][EAP7-1192] Hard requirements

### PR DESCRIPTION
**Pull request by tadamski**

### CODE REVIEW

File: openshift/CLOUD-2262.adoc [hunk 1]
```diff
@@ -258,11 +258,17 @@ This should be considered as a new feature request.
...
+* applications should be able to communicate using ejb-client/remoting libraries
+* distributed transactional operations among those applications should be fully supported
+* transactions should recover properly if the transaction is interrupted
+* users would create the applications using provided template, which would hide the complexity associated with operation in cloud
+* users would be able to configure connections between applications by configuring remoting subsystem in standalone-openshift.xml
+
 
 === Nice-to-Have Requirements
+* scale-down of number of replicas in StatefulSet - this is planed to be introduced as en extension after the core functionality is implemented and tested
+* auto-generate standalone-openshift.xml during the config - this is planed to be introduced as en extension after the core functionality is implemented and teste
```
Comment (ochaloup) [inline openshift/CLOUD-2262.adoc]:
```suggestion
* auto-generate `standalone-openshift.xml` during the config - this is planned to be introduced as en extension after the core functionality is implemented and tested
```

Comment (ochaloup) [inline openshift/CLOUD-2262.adoc]: what does mean `auto-generate` the `standalone-openshift.xml`?

File: openshift/CLOUD-2262.adoc [hunk 2]
```diff
@@ -258,11 +258,17 @@ This should be considered as a new feature request.
...
 
-* EJB remote calls have to work when application is deployed on JBoss EAP on OpenShift
-* The transaction consistency has to be guaranteed for the transactions propagated
-  over the EJB remote calls
+* allow users to deploy clustered transactional EAP applications on OpenShift
+* applications should be able to communicate using ejb-client/remoting libraries
+* distributed transactional operations among those applications should be fully supported
+* transactions should recover properly if the transaction is interrupted
+* users would create the applications using provided template, which would hide the complexity associated with operation in cloud
+* users would be able to configure connections between applications by configuring remoting subsystem in standalone-openshift.xml
```
Comment (ochaloup) [inline openshift/CLOUD-2262.adoc]: will we define the ways how the remote connection can be created? will we be supporting only remote outbound connection at this time?
As we discussed the recovery with the programmatic lookup does not work currently. Do you expect that being nice to have or hard requirement?

File: openshift/CLOUD-2262.adoc [hunk 3]
```diff
@@ -258,11 +258,17 @@ This should be considered as a new feature request.
 
 === Hard Requirements
 
-* EJB remote calls have to work when application is deployed on JBoss EAP on OpenShift
-* The transaction consistency has to be guaranteed for the transactions propagated
-  over the EJB remote calls
+* allow users to deploy clustered transactional EAP applications on OpenShift
```
Comment (ochaloup) [inline openshift/CLOUD-2262.adoc:261]: will we define in the requirements that the stateful set will be used or is it rather an implementation detail?

**Review by ochaloup — COMMENTED** [review]
@tadamski I added few comments for discussions. Thanks for the submitting this.

File: openshift/CLOUD-2262.adoc (hunk 3 above)
Comment (ochaloup) [reply openshift/CLOUD-2262.adoc:261]: not needed  :)

File: openshift/CLOUD-2262.adoc (hunk 1 above)
Comment (tadamski) [reply openshift/CLOUD-2262.adoc]: This has to be still designed. Possibly by scripting the template. I would like to decouple OpenShift internals from customers as much as possible.

Comment (tadamski) [reply openshift/CLOUD-2262.adoc]: fixed

File: openshift/CLOUD-2262.adoc (hunk 2 above)
Comment (tadamski) [reply openshift/CLOUD-2262.adoc]: I added programmatic lookup as 'nice-to-have' cause we don't know the reason for the problems yet.

**Pull request merged**